R/subscript.R

```diff
@@ -64,36 +64,42 @@ vec_as_subscript_result <- function(i,
 #' @export
 vec_as_subscript2 <- function(i,
                               ...,
-                              logical = c("cast", "error"),
                               numeric = c("cast", "error"),
                               character = c("cast", "error"),
                               arg = NULL,
                               call = caller_env()) {
-  check_dots_empty0(...)
+  check_dots <- function(..., logical = "error") {
+    if (!is_string(logical, "error")) {
+      abort(
+        "`vctrs::vec_as_subscript2(logical = 'cast')` is deprecated.",
+        call = caller_env()
+      )
+    }
+    check_dots_empty0(...)
```

Comment:
Do we want to tweak the `call` of `check_dots_empty0()` to also be `caller_env()`?

Reply:
yup good catch